In [1]:
#importação das bibliotecas

import pandas as pd
import numpy as np

In [3]:
#Adição da base de dados

base = pd.read_csv("base_de_dados/iris.csv", sep=",")
base.shape #Visualizando a quantidade de linhas e colunas

(150, 5)

In [6]:
base.head(10)

sepal length  sepal width  petal length  petal width        class
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa
5           5.4          3.9           1.7          0.4  Iris-setosa
6           4.6          3.4           1.4          0.3  Iris-setosa
7           5.0          3.4           1.5          0.2  Iris-setosa
8           4.4          2.9           1.4          0.2  Iris-setosa
9           4.9          3.1           1.5          0.1  Iris-setosa

In [8]:
np.random.seed(2743) #Definindo uma semente

valores = np.random.choice(a = [0, 1], size = len(base), replace = True, p = [0.7, 0.3])

A variável valores vai receber um array com um tamanho de 150 que é igual ao len(base), esse array será composto pelos números 0 e 1 que são as opções para serem escolhidas (a = [0, 1]),esses dois números poderão ser repitidos mais de uma vez pois o replace foi definido como True o que permite a reposição dos dados e por último definimos que o zero tera 0.7 (70%) de chance de ser selecionado e o um 0.3 (30%) de chance de ser selecionado.

In [11]:
valores

array([0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1,
       0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1,
       0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1,
       0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0,
       0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0,
       0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0])

In [13]:
len(valores)

150

In [15]:
amostra = base.loc[valores == 1]

amostra.shape

(49, 5)

In [23]:
amostra

sepal length  sepal width  petal length  petal width            class
4             5.0          3.6           1.4          0.2      Iris-setosa
7             5.0          3.4           1.5          0.2      Iris-setosa
10            5.4          3.7           1.5          0.2      Iris-setosa
12            4.8          3.0           1.4          0.1      Iris-setosa
15            5.7          4.4           1.5          0.4      Iris-setosa
16            5.4          3.9           1.3          0.4      Iris-setosa
17            5.1          3.5           1.4          0.3      Iris-setosa
21            5.1          3.7           1.5          0.4      Iris-setosa
24            4.8          3.4           1.9          0.2      Iris-setosa
28            5.2          3.4           1.4          0.2      Iris-setosa
32            5.2          4.1           1.5          0.1      Iris-setosa
34            4.9          3.1           1.5          0.1      Iris-setosa
35            5.0          3.2           1.2          0.2      Iris-setosa
39            5.1          3.4           1.5          0.2      Iris-setosa
43            5.0          3.5           1.6          0.6      Iris-setosa
46            5.1          3.8           1.6          0.2      Iris-setosa
47            4.6          3.2           1.4          0.2      Iris-setosa
48            5.3          3.7           1.5          0.2      Iris-setosa
52            6.9          3.1           4.9          1.5  Iris-versicolor
59            5.2          2.7           3.9          1.4  Iris-versicolor
62            6.0          2.2           4.0          1.0  Iris-versicolor
65            6.7          3.1           4.4          1.4  Iris-versicolor
69            5.6          2.5           3.9          1.1  Iris-versicolor
74            6.4          2.9           4.3          1.3  Iris-versicolor
75            6.6          3.0           4.4          1.4  Iris-versicolor
80            5.5          2.4           3.8          1.1  Iris-versicolor
82            5.8          2.7           3.9          1.2  Iris-versicolor
84            5.4          3.0           4.5          1.5  Iris-versicolor
89            5.5          2.5           4.0          1.3  Iris-versicolor
92            5.8          2.6           4.0          1.2  Iris-versicolor
93            5.0          2.3           3.3          1.0  Iris-versicolor
94            5.6          2.7           4.2          1.3  Iris-versicolor
99            5.7          2.8           4.1          1.3  Iris-versicolor
103           6.3          2.9           5.6          1.8   Iris-virginica
106           4.9          2.5           4.5          1.7   Iris-virginica
111           6.4          2.7           5.3          1.9   Iris-virginica
115           6.4          3.2           5.3          2.3   Iris-virginica
116           6.5          3.0           5.5          1.8   Iris-virginica
117           7.7          3.8           6.7          2.2   Iris-virginica
121           5.6          2.8           4.9          2.0   Iris-virginica
134           6.1          2.6           5.6          1.4   Iris-virginica
136           6.3          3.4           5.6          2.4   Iris-virginica
138           6.0          3.0           4.8          1.8   Iris-virginica
139           6.9          3.1           5.4          2.1   Iris-virginica
140           6.7          3.1           5.6          2.4   Iris-virginica
143           6.8          3.2           5.9          2.3   Iris-virginica
144           6.7          3.3           5.7          2.5   Iris-virginica
146           6.3          2.5           5.0          1.9   Iris-virginica
147           6.5          3.0           5.2          2.0   Iris-virginica

In [21]:
#Outra forma para também gerar a amostra simples

tamanho_amostra = 40

novos_valores = np.random.choice(a = base.index, size = tamanho_amostra, replace = False)

amostra2 = base.loc[novos_valores]

amostra2

sepal length  sepal width  petal length  petal width            class
79            5.7          2.6           3.5          1.0  Iris-versicolor
38            4.4          3.0           1.3          0.2      Iris-setosa
96            5.7          2.9           4.2          1.3  Iris-versicolor
48            5.3          3.7           1.5          0.2      Iris-setosa
77            6.7          3.0           5.0          1.7  Iris-versicolor
14            5.8          4.0           1.2          0.2      Iris-setosa
45            4.8          3.0           1.4          0.3      Iris-setosa
13            4.3          3.0           1.1          0.1      Iris-setosa
107           7.3          2.9           6.3          1.8   Iris-virginica
127           6.1          3.0           4.9          1.8   Iris-virginica
71            6.1          2.8           4.0          1.3  Iris-versicolor
146           6.3          2.5           5.0          1.9   Iris-virginica
136           6.3          3.4           5.6          2.4   Iris-virginica
15            5.7          4.4           1.5          0.4      Iris-setosa
82            5.8          2.7           3.9          1.2  Iris-versicolor
28            5.2          3.4           1.4          0.2      Iris-setosa
111           6.4          2.7           5.3          1.9   Iris-virginica
66            5.6          3.0           4.5          1.5  Iris-versicolor
3             4.6          3.1           1.5          0.2      Iris-setosa
30            4.8          3.1           1.6          0.2      Iris-setosa
69            5.6          2.5           3.9          1.1  Iris-versicolor
110           6.5          3.2           5.1          2.0   Iris-virginica
17            5.1          3.5           1.4          0.3      Iris-setosa
99            5.7          2.8           4.1          1.3  Iris-versicolor
70            5.9          3.2           4.8          1.8  Iris-versicolor
7             5.0          3.4           1.5          0.2      Iris-setosa
87            6.3          2.3           4.4          1.3  Iris-versicolor
46            5.1          3.8           1.6          0.2      Iris-setosa
147           6.5          3.0           5.2          2.0   Iris-virginica
24            4.8          3.4           1.9          0.2      Iris-setosa
31            5.4          3.4           1.5          0.4      Iris-setosa
32            5.2          4.1           1.5          0.1      Iris-setosa
47            4.6          3.2           1.4          0.2      Iris-setosa
115           6.4          3.2           5.3          2.3   Iris-virginica
80            5.5          2.4           3.8          1.1  Iris-versicolor
89            5.5          2.5           4.0          1.3  Iris-versicolor
67            5.8          2.7           4.1          1.0  Iris-versicolor
120           6.9          3.2           5.7          2.3   Iris-virginica
88            5.6          3.0           4.1          1.3  Iris-versicolor
108           6.7          2.5           5.8          1.8   Iris-virginica